src/ffs/pub/tree.ts

```diff
@@ -0,0 +1,78 @@
+import pathUtil from '../path'
+import publicNode, { PublicNode } from "./node"
+import { Tree, Link } from "../types"
+import { CID, FileContent } from '../../ipfs'
+import { getRecurse, addChildRecurse } from "../helpers"
+
+export class PublicTree implements Tree {
+
+  root: PublicNode
+
+  constructor(root: PublicNode) {
+    this.root = root
+  }
+
+  async put(): Promise<CID> {
+    return this.root.put()
+  }
+
+  async cid(): Promise<CID> {
+    return this.put()
+  }
+
+  async listDir(path: string): Promise<Link[] | null> {
+    const node = await this.getNode(path)
+    return node?.links() || []
```

Comment:
Would it be possible to use https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/FILES.md#ls here instead, so we can get the `mtime` (last-modified timestamp)? Or is there another way to get that? Also, should I run `ffs.pubNode.isFile` on each of these links to find out if I'm dealing with a directory or a file? (`files.ls` has a `type` property for this)

Reply:
yeah totally. Sorry didn't realize these were being used! I'll modify our `Link` type to include `mtime: number` & `isFile: boolean` and add these values to public nodes using `ipfs.ls` and include the data in our uploaded encrypted nodes as well :+1: